dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -95,14 +95,14 @@ MPPTunnelBase<Writer>::~MPPTunnelBase()
         }
         LOG_FMT_TRACE(log, "waiting consumer finish!");
         waitForConsumerFinish(/*allow_throw=*/false);
+        LOG_FMT_TRACE(log, "waiting child thread finished!");
+        thread_manager->wait();
```

Comment:
if throw exceptions before this line, the wait() would miss?

Reply:
But the only case an exception may be thrown is that timeout is hit in `waitUntilConnectedOrFinished`, which mean the tunnel is not connected, if the tunnel is not connected, we don't need to wait  the sending thread since it only starts after the tunnel is connected.

Reply:
ok